knox/README.md

```diff
@@ -0,0 +1,103 @@
+# Apache Knox Initialization Action
+
+This [initialization action](https://cloud.google.com/dataproc/init-actions)
+installs [Apache Knox](https://knox.apache.org/) on the first master node within a
+[Google Cloud Dataproc](https://cloud.google.com/dataproc) cluster. The Knox distributed
+by the BigTop project is installed via `apt-get` so it may not be the latest Knox.
+
+Moreover, initialization action configures the knox gateway by:
+- adding topologies for backend clusters
+- adding or creating TLS certificates for HTTPS
+- setting the master key of Knox
+- altering the main configuration file, *gateway-site.xml*
+- adds a cron schedule to update the configuration from a GCS bucket periodically
+
+
+### Caveats
+
+ - Dataproc Jobs API uses binary transport protocol of Hive. When we switch the protocol to HTTP for Knox, you cannot use the Jobs API for these backend clusters. See [Additional setup for Hive](#additional-setup-for-hive) for details.
+ - [Component Gateway](https://cloud.google.com/dataproc/docs/concepts/accessing/dataproc-gateways) does not work for the gateway cluster when *SSL* is enabled by changing [gateway-site.xml](./gateway-site.xml) or when [default.xml](./topologies/default.xml) is altered since component gateway also uses Knox. Note that, the backend clusters that gateway redirects can still use the component gateway.
+
+## Using this initialization action
+
+This initialization action requires a bucket that stores configuration. By providing a bucket with configurations, you can create a Dataproc cluster with Knox installed as follows:
+
```

Comment:
Add disclaimer/notice about using init actions in prod (see other init action docs)

Reply:
done.